lib/query_components/filter.rb

```diff
@@ -1,23 +1,27 @@
+require "query_components/user_filter"
+require "query_components/visibility_filter"
+require 'elasticsearch/queries'
+
 module QueryComponents
   class Filter < BaseComponent
+    include Elasticsearch::Queries
+
     def payload(excluding = [])
-      rejects = []
-      filters = []
+      user_filter = QueryComponents::UserFilter.new(search_params)
+      visibility_filter = QueryComponents::VisibilityFilter.new(search_params)
 
-      param_filters = search_params.filters.reject do |filter|
-        excluding.include?(filter.field_name)
-      end
+      filters = combine_filters(user_filter.filters_payload(excluding), :and)
```

Comment:
for consistency, please rewrite this with the same style you've used in the lines below